reflex/components/core/match.py

```diff
@@ -0,0 +1,243 @@
+
+        Raises:
+            ValueError: When a default case is not provided for cases with Var return types.
+        """
+        match_cond_var = cls._create_condition_var(cond)
+        cases, default = cls._process_cases(list(cases))
+        match_cases = cls._process_match_cases(cases)
+
+        cls._validate_return_types(match_cases)
+
+        if default is None and types._issubclass(type(match_cases[0][-1]), BaseVar):
+            raise ValueError(
+                "For cases with return types as Vars, a default case must be provided"
+            )
+
+        return cls._create_match_cond_var_or_component(
+            match_cond_var, match_cases, default
+        )
+
+    @classmethod
+    def _create_condition_var(cls, cond: Any) -> BaseVar:
+        """Convert the condition to a Var.
+
+        Args:
+            cond: The condition.
+
+        Returns:
+            The condition as a base var
+
+        Raises:
+            ValueError: If the condition is not provided.
+        """
+        match_cond_var = Var.create(cond)
+        if match_cond_var is None:
+            raise ValueError("The condition must be set")
+        return match_cond_var  # type: ignore
+
+    @classmethod
+    def _process_cases(
+        cls, cases: List
+    ) -> Tuple[List, Optional[Union[BaseVar, BaseComponent]]]:
+        """Process the list of match cases and the catchall default case.
+
+        Args:
+            cases: The list of match cases.
+
+        Returns:
+            The default case and the list of match case tuples.
+
+        Raises:
+            ValueError: If there are multiple default cases.
+        """
+        default = None
+
+        if len([case for case in cases if not isinstance(case, tuple)]) > 1:
+            raise ValueError("rx.match can only have one default case.")
+
+        # Get the default case which should be the last non-tuple arg
+        if not isinstance(cases[-1], tuple):
+            default = cases.pop()
+            default = (
+                Var.create(default)
+                if not isinstance(default, BaseComponent)
+                else default
+            )
+
+        return cases, default  # type: ignore
+
+    @classmethod
+    def _process_match_cases(cls, cases: List) -> List[List[BaseVar]]:
+        """Process the individual match cases.
+
+        Args:
+            cases: The match cases.
+
+        Returns:
+            The processed match cases.
+
+        Raises:
+            ValueError: If the default case is not the last case or the tuple elements are less than 2.
+        """
+        match_cases = []
+        for case in cases:
+            if not isinstance(case, tuple):
+                raise ValueError(
+                    "rx.match should have tuples of cases and a default case as the last argument."
+                )
+            # There should be at least two elements in a case tuple(a condition and return value)
+            if len(case) < 2:
```

Comment:
shouldn't this `if len(case) != 2`... i'd consider it a programming error if the case tuple has multiple return values, unless i'm missing something in the implementation.

Reply:
The first `n-1` elements in the tuple are treated as the cases to be matched and the last value as the return value, which means there can only be one return value in a tuple. A minimum of two elements means, the first value is treated as the match case the second as the return value

Reply:
ohhh yeah that's right; we talked about that.... we should make sure that's clearly documented though so people know what to expect